pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,129 @@
+	// This field replaces MachineCIDR, and if set MachineCIDR must
+	// be empty or match the first entry in the list.
+	// Default is 10.0.0.0/16 for all platforms other than libvirt.
+	// For libvirt, the default is 192.168.126.0/24.
+	//
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+
+	// ClusterNetwork is the list of IP address pools for pods.
+	// Default is 10.128.0.0/14 and a host prefix of /23.
+	//
+	// +optional
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork,omitempty"`
+
+	// ServiceNetwork is the list of IP address pools for services.
+	// Default is 172.30.0.0/16.
+	// NOTE: currently only one entry is supported.
+	//
+	// +kubebuilder:validation:MaxItems=1
+	// +optional
+	ServiceNetwork []string `json:"serviceNetwork,omitempty"`
+}
+
+// MachineNetworkEntry is a single IP address block for node IP blocks.
+type MachineNetworkEntry struct {
+	// CIDR is the IP block address pool for machines within the cluster.
+	CIDR string `json:"cidr"`
+}
+
+// ClusterNetworkEntry is a single IP address block for pod IP blocks. IP blocks
+// are allocated with size 2^HostSubnetLength.
+type ClusterNetworkEntry struct {
+	// CIDR is the IP block address pool.
+	CIDR string `json:"cidr"`
+
+	// HostPrefix is the prefix size to allocate to each node from the CIDR.
+	// For example, 24 would allocate 2^8=256 adresses to each node. If this
+	// field is not used by the plugin, it can be left unset.
+	// +optional
+	HostPrefix int32 `json:"hostPrefix,omitempty"`
+}
+
+type ClusterProgressInfo struct {
+	// progress info
+	ProgressInfo string `json:"progressInfo,omitempty"`
+	// Time at which the cluster install progress was last updated.
+	LastProgressUpdateTime *metav1.Time `json:"lastProgressUpdateTime,omitempty"`
+}
+
+type HostNetwork struct {
+	// cidr
+	Cidr string `json:"cidr,omitempty"`
+	// host ids
+	HostIds []string `json:"hostIDs"`
+}
+
+// ClusterStatus defines the observed state of Cluster
+type ClusterStatus struct {
+	State                        string              `json:"state,omitempty"`
+	StateInfo                    string              `json:"stateInfo,omitempty"`
+	HostNetworks                 []HostNetwork       `json:"hostNetworks,omitempty"`
+	InstallationStartTime        string              `json:"installationStartTime,omitempty"`
+	InstallationCompletionTime   *metav1.Time        `json:"installationCompletionTime,omitempty"`
+	Hosts                        int                 `json:"hosts,omitempty"`
+	Progress                     ClusterProgressInfo `json:"progress,omitempty"`
+	ValidationsInfo              string              `json:"validationsInfo,omitempty"`
+	ConnectivityMajorityGroups   string              `json:"connectivityMajorityGroups,omitempty"`
+	LastUpdateTime               *metav1.Time        `json:"lastUpdateTime,omitempty"`
+	ControllerLogsCollectionTime *metav1.Time        `json:"controllerLogsCollectionTime,omitempty"`
```

Comment:
What controller logs is this referring to, and where are they collected?

Reply:
This is a bad name.  Once we have the two-node control plane and are waiting for the third master to pivot, we run a k8s job to monitor and report the remainder of the progress (because at this point we have no more agents running).  So here we include the logs of the job itself (e.g. waiting for the 3rd master to join, waiting for console, waiting for CVO), ad well as the output of `oc must-gather`.

Reply:
We suspect this would be covered by the LogsCollected condition's LastProbeTime if state is true. This is mostly just used for debugging.

Will drop this property.